Userland/Libraries/LibWeb/TreeNode.h

```diff
@@ -97,6 +97,33 @@ class TreeNode : public Weakable<T> {
         return const_cast<TreeNode*>(this)->child_at_index(index);
     }
 
+    int child_index(const T& search_child)
+    {
+        int index = 0;
+        auto* child = first_child();
```

Comment:
There may not be any children, which will cause first_child to return nullptr and cause a null deref when trying to get the next sibling in the while loop.

Reply:
Using Kling's `VERIFY(search_child.parent() == this)` from above would mitigate this as we check whether we are the parent of the given node and therefore we have at least one child.
But in case something fishy is going on, we can still check for child being null in the while ;)

Reply:
If we're asserting that, we could also assert that child isn't null, since like you said that assertion means we have at least one child.

Reply:
Thats probably a good idea too!